src/couch_epi_data_source.erl

```diff
@@ -0,0 +1,173 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+% http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_epi_data_source).
+
+-behaviour(gen_server).
+-define(MONITOR_INTERVAL, 5000).
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([start_link/4, reload/1]).
+-export([wait/1, stop/1]).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Exports
+%% ------------------------------------------------------------------
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+-record(state, {
+    subscriber, locator, key, hash, handle,
+    initialized = false, pending = []}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(SubscriberApp, {epi_key, Key}, Src, Options) ->
+    {ok, Locator} = locate(SubscriberApp, Src),
+    gen_server:start_link(?MODULE, [SubscriberApp, Locator, Key, Options], []).
+
+reload(Server) ->
+    gen_server:call(Server, reload).
+
+wait(Server) ->
```

Comment:
Does anything call the wait functions? I see the fancy delayed init but I'm not sure I understand the reasoning for it.